src/core/types.ts

```diff
@@ -0,0 +1,109 @@
+
```

Comment:
Note sure whether we should have a dedicated file for types. Probably better to put these in files together with classes they relate to.